core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -133,7 +133,6 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
     Utils.getConfiguredLocalDirs(conf).flatMap { rootDir =>
       try {
         val localDir = Utils.createDirectory(rootDir, "blockmgr")
-        Utils.chmod700(localDir)
```

Comment:
No. The cluster manager is responsible for securing the app's directory (in YARN, only the app owner and the YARN group can read it; in standalone, only the user running the standalone service can; not sure about Mesos, which is why the Mesos code might need to be fixed separately).